3scaleAdapter/pkg/threescale/threescale.go

```diff
@@ -50,6 +49,110 @@ type (
 // For this PoC I'm using the authorize template, but we should check if the quota template
 // is more convenient and we can do some optimizations around.
 var _ authorization.HandleAuthorizationServiceServer = &Threescale{}
+var _ logentry.HandleLogEntryServiceServer = &Threescale{}
+
+func (s *Threescale) HandleLogEntry(ctx context.Context, l *logentry.HandleLogEntryRequest) (*v1beta1.ReportResult, error) {
+	var report v1beta1.ReportResult
+	var err error
+
+	if l.AdapterConfig == nil {
+		return &report, fmt.Errorf("adapter config not available")
+	}
+
+	cfg := &config.Params{}
+	if err := cfg.Unmarshal(l.AdapterConfig.Value); err != nil {
+		log.Errorf("error unmarshalling adapter config: %v", err)
+		return &report, err
+	}
+
+	systemC, err := s.systemClientBuilder(cfg.SystemUrl)
+
+	if err != nil {
+		log.Errorf("error building HTTP client for 3scale system - %s", err.Error())
+		return &report, err
+	}
+
+	err = s.ReportUsage(cfg, l.Instances, systemC)
+
+	if err != nil {
+		log.Errorf("error reporting usage - %s", err.Error())
+	}
+
+	return &report, err
```

Comment:
I think this function should use the istio `status` package to correctly return the errors with some message that gives context

Reply:
To implement the interface, we need to return a report (this is a report type of template, not a check template)